src/avalanchemq/queue/queue.cr

```diff
@@ -825,6 +826,7 @@ module AvalancheMQ
       count = @ready.purge
       @log.debug { "Purged #{count} messages" }
       @vhost.dirty = true
+      @vhost.gc_loop.send nil # trigger gc for message clean up
```

Comment:
how about a new method called `trigger_gc` that sets the dirty flag and sends nil to the channel?